lib/SOIL2/CMakeLists.txt

```diff
@@ -4,6 +4,7 @@ project(SOIL2)
 
 if (NOT WIN32)
   add_definitions(-fPIC)
+  set(CMAKE_POSITION_INDEPENDENT_CODE 1)
```

Comment:
this handles -fPIC where necessary and nothing on win32 so you don't need the if.